pkg/controllers/workapplier/controller_integration_test.go

```diff
@@ -3952,3 +3952,615 @@ var _ = Describe("report diff", func() {
 		})
 	})
 })
+
+var _ = Describe("switch apply strategies", func() {
+	Context("switch from report diff to CSA", Ordered, func() {
+		workName := fmt.Sprintf(workNameTemplate, "d1")
+		// The environment prepared by the envtest package does not support namespace
+		// deletion; each test case would use a new namespace.
+		nsName := fmt.Sprintf(nsNameTemplate, "d1")
+		deployName := fmt.Sprintf(deployNameTemplate, "d1")
+
+		var appliedWorkOwnerRef *metav1.OwnerReference
+		var regularNS *corev1.Namespace
+		var regularDeploy *appsv1.Deployment
+
+		BeforeAll(func() {
+			// Prepare a NS object.
+			regularNS = ns.DeepCopy()
+			regularNS.Name = nsName
+			regularNSJSON := marshalK8sObjJSON(regularNS)
+
+			// Prepare a Deployment object.
+			regularDeploy = deploy.DeepCopy()
+			regularDeploy.Namespace = nsName
+			regularDeploy.Name = deployName
+			regularDeployJSON := marshalK8sObjJSON(regularDeploy)
+
+			// Create the objects first in the member cluster.
+			Expect(memberClient.Create(ctx, regularNS)).To(Succeed(), "Failed to create the NS object")
+
+			regularDeploy.Spec.Replicas = ptr.To(int32(2))
```

Comment:
add a comment to highlight that we introduced the diff here?

Reply:
Set to 10 seconds. With 5, things are a bit flaky on my local machine 😣